query-node/mappings/src/content/channel.ts

```diff
@@ -0,0 +1,313 @@
+import { fixBlockTimestamp } from '../eventFix'
+import { SubstrateEvent } from '@dzlzv/hydra-common'
+import { DatabaseManager } from '@dzlzv/hydra-db-utils'
+import ISO6391 from 'iso-639-1';
+import { FindConditions, In } from 'typeorm'
+
+import { AccountId } from "@polkadot/types/interfaces";
+import { Option } from '@polkadot/types/codec';
+import { Content } from '../../../generated/types'
+import {
+  readProtobuf,
+  readProtobufWithAssets,
+  convertContentActorToOwner,
+} from './utils'
+
+import {
+  Channel,
+  ChannelCategory,
+  DataObject,
+} from 'query-node'
+import {
+  inconsistentState,
+  logger,
+} from '../common'
+
+import {
+  AssetAvailability,
+} from 'query-node'
+import BN from 'bn.js'
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export async function content_ChannelCreated(db: DatabaseManager, event: SubstrateEvent): Promise<void> {
+  // read event data
+  const {channelId, channelCreationParameters, contentActor} = new Content.ChannelCreatedEvent(event).data
+
+  // read metadata
+  const protobufContent = await readProtobufWithAssets(
+    new Channel(),
+    {
+      metadata: channelCreationParameters.meta,
+      db,
+      blockNumber: event.blockNumber,
+      assets: channelCreationParameters.assets,
+      contentOwner: convertContentActorToOwner(contentActor, channelId.toBn()),
+    }
+  )
+
+  // create entity
+  const channel = new Channel({
+    // main data
+    id: channelId.toString(),
```

Comment:
Why are we using string representation for a number?
Shouldn't this be channelId.toNumber()?

Reply:
Schema contains this definition of id: 
```
type Channel @entity {
  "Runtime entity identifier (EntityId)"
  id: ID!
...
}
```
The `ID` translates to `string` field in database. That's why there is a conversion to `string`. `toNumber()` would cause compilation error.